src/components/Excelerator/Excelerator.js

```diff
@@ -19,52 +47,85 @@ import _ from "lodash";
         }
     ]
  */
-export function loadDataFromExcelFile(excel) {
-  const workbook = XLSX.read(excel, {
-    type: "binary"
-  });
+export function loadDataFromExcelFile(entitySchema, excel) {
+  const workbook = XLSX.read(excel, { type: "binary" });
 
-  const [mainType, ...listsNames] = workbook.SheetNames;
+  const [mainSheetName, ...listsNames] = workbook.SheetNames;
 
-  const data = XLSX.utils.sheet_to_json(workbook.Sheets[mainType]);
+  const mainSheetData = XLSX.utils.sheet_to_json(workbook.Sheets[mainSheetName]);
+  //example: array of users
+  const mainEntitiesArr = loadMainSheetData(entitySchema, mainSheetData);
 
-  if (listsNames) {
-    addListsToMainTypeObject(data, listsNames, workbook);
-  }
+  listsNames.forEach(ln => {
+    let listSheetData = XLSX.utils.sheet_to_json(workbook.Sheets[ln]);
+    if (entitySchema.scalarListsNames.includes(ln)) {
+      addScalarListToMainEntitiesArr(mainEntitiesArr, entitySchema, ln, listSheetData);
+    } else {
+      addObjListToMainEntitiesArr(mainEntitiesArr, entitySchema, ln, listSheetData);
+    }
+  })
 
-  return data;
+  return mainEntitiesArr;
 }
 
-/**
- *
- * @param {*} mainData The main type. For example: "User"
- * @param {*} listNames The names of the lists connected to the main type. For example: ["Children", "Certifications"]
- * @param {*} workbook The workbook in which the data is saved. First sheet is the main type and the rest are the lists. Can be acquired using XLSX.utils.book_new()
- * @description Addsd the list data to the main type
- */
-function addListsToMainTypeObject(mainData, listNames, workbook) {
-  listNames.forEach(name => {
-    // listData = [{id:5, name:"Bob", parentId: 1},{id:6, name:"John", parentId: 1}]
-    const listData = XLSX.utils.sheet_to_json(workbook.Sheets[name]);
-    const dataGroupedByParentId = _.groupBy(listData, x => x.parentId);
+function loadMainSheetData(entitySchema, sheetData) {
+  const mainEntitiesArr = sheetData.map(entryDataFromExcel => {
+    let entry = {};
+
+    entitySchema.scalarFields.forEach(sf => {
+      entry[`${sf}`] = entryDataFromExcel[`${sf}`]
+    });
+
+    entitySchema.embeddedObjsNames.forEach(eon => {
+      entry[`${eon}`] = {};
+
+      entitySchema[`${eon}`].scalarFields.forEach(eosf => {
+        entry[`${eon}`][`${eosf}`] = entryDataFromExcel[`${eon}.${eosf}`];
+      })
+    })
+
+    return entry;
+  })
+
+  return mainEntitiesArr;
+
+}
 
-    _.keys(dataGroupedByParentId).forEach(parentId => {
-      const parent = mainData.find(entry => entry.id === parentId);
+function addScalarListToMainEntitiesArr(mainEntitiesArr, entitySchema, listName, listSheetData) {
+  mainEntitiesArr.forEach(me => {
+    me[`${listName}`] = [];
 
-      if (!parent) {
-        throw new Error(`Parent with id ${parentId} cannot be null`);
+    listSheetData.forEach(lsde => {
+      if (lsde[`${entitySchema.entityName}Id`] === me.id){
```

Comment:
maybe extract `${entitySchema.entityName}Id` to variable so it will be more readble and call it parentId or something like that